templates/content/spark.core.site.conf.tt

```diff
@@ -0,0 +1,25 @@
+{{contenttype "application/xml" -}}
+<configuration>
+  <property>
+      <name>fs.s3a.path.style.access</name>
+      <value>true</value>
+  </property>
+  <property>
+      <name>fs.s3a.impl</name>
+      <value>org.apache.hadoop.fs.s3a.S3AFileSystem</value>
+  </property>
+  <property>
+      <name>fs.s3a.endpoint</name>
+      <value>{{ .Query.lakefs_url }}</value>
+  </property>
+{{- with $creds := new_credentials "spark"}}
+  <property>
+      <name>fs.s3a.access.key</name>
```

Comment:
Wondering if we want to use per-bucket configuration here, with the repository name.
Pros: less intrusive.
Cons: may bot work with older Hadoop versions.

Leaning towards yes. WDYT?

Reply:
Thought about it, yet as you said, if the users have a Hadoop version older than 2.8, the configurations won't work at all.
Do you have details about the Hadoop API versions users use?

Reply:
I don't have this information. However hadoop-aws:2.7.7 is 4 years old. Not sure we have to support it, leaving it for your consideration. If we get rejections in the future we can adapt.